nixos/modules/system/boot/resolved.nix

```
@@ -214,6 +269,21 @@ in
 
       networking.resolvconf.package = pkgs.systemd;
 
+      networking.firewall =
+        let
+          mkZeroConf = option: option.enable && option.openFirewall;
+          mkZeroConfDeprecated =
+            option: if builtins.elem option zeroConfDeprecatedOpts then false else mkZeroConf option;
+        in
+        lib.mkMerge [
+          (mkIf (mkZeroConf cfg.mdns) {
+            allowedUDPPorts = [ 5353 ];
+          })
+
+          (mkIf (mkZeroConfDeprecated cfg.llmnr) {
+            allowedUDPPorts = [ 5355 ];
+          })
+        ];
```

This can be simplified a bunch:
```suggestion
      networking.firewall = let
        fromAttrs = { enable, openFirewall, ... }: enable && openFirewall;
      in [ {
        allowedUDPPorts =
          (lib.optional (fromAttrs cfg.mdns) 5353)
          ++ (lib.optional (lib.isAttrs cfg.llmnr && fromAttrs cfg.llmnr) 5355);
      } ];
```

---

Looks like this suggestion was partially incorrect, pushed a fix.